strawberryfields/engine.py

```diff
@@ -575,19 +582,36 @@ def target(self) -> str:
         return self._target
 
     @property
-    def connection(self) -> Connection:
-        """The connection object used by the engine.
+    def connection(self) -> xcc.Connection:
+        """The connection used by the engine. A new :class:`xcc.Connection` will
+        be created and returned each time this property is accessed if the
+        connection supplied to :meth:`~.RemoteEngine.__init__` was ``None``.
 
         Returns:
-            strawberryfields.api.Connection
+            xcc.Connection
         """
-        return self._connection
+        if self._connection is not None:
+            return self._connection
+
+        settings = xcc.Settings()
+
+        return xcc.Connection(
+            refresh_token=settings.REFRESH_TOKEN,
+            access_token=settings.ACCESS_TOKEN,
+            host=settings.HOST,
+            port=settings.PORT,
+            tls=settings.TLS,
+            headers={"User-Agent": f"StrawberryFields/{__version__}"},
+        )
 
     @property
     def device_spec(self):
         """The device specifications for target device"""
         if self._spec is None:
-            self._spec = self._connection.get_device_spec(self.target)
+            target = self.target
+            connection = self.connection
+            device = xcc.Device(target=target, connection=connection)
+            self._spec = DeviceSpec(target=target, connection=connection, spec=device.specification)
         return self._spec
 
     def run(
```

Comment:
What do you think about adding a `Raises:` section to the docstring?

Reply:
Done.